channel/buffs.go

```diff
@@ -0,0 +1,647 @@
+
+	// 2nd Job - Warrior branches
+	AddSkillBuff(int32(skill.SwordBooster), BuffBooster)
+	AddSkillBuff(int32(skill.AxeBooster), BuffBooster)
+	AddSkillBuff(int32(skill.PageSwordBooster), BuffBooster)
+	AddSkillBuff(int32(skill.BwBooster), BuffBooster)
+	AddSkillBuff(int32(skill.SpearBooster), BuffBooster)
+	AddSkillBuff(int32(skill.PolearmBooster), BuffBooster)
+
+	AddSkillBuff(int32(skill.Rage), BuffWeaponAttack)
+	AddSkillBuff(int32(skill.PowerGuard), BuffPowerGuard)
+	AddSkillBuff(int32(skill.PagePowerGuard), BuffPowerGuard)
+
+	AddSkillBuff(int32(skill.IronWill), BuffWeaponDefense, BuffMagicDefense)
+	AddSkillBuff(int32(skill.HyperBody), BuffMaxHP, BuffMaxMP)
+
+	// 2nd Job - Magician branches
+	AddSkillBuff(int32(skill.Meditation), BuffMagicAttack)
+	AddSkillBuff(int32(skill.ILMeditation), BuffMagicAttack)
+	AddSkillBuff(int32(skill.Invincible), BuffInvincible)
+
+	// 2nd Job - Archer branches
+	AddSkillBuff(int32(skill.BowBooster), BuffBooster)
+	AddSkillBuff(int32(skill.CrossbowBooster), BuffBooster)
+	AddSkillBuff(int32(skill.SoulArrow), BuffSoulArrow)
+	AddSkillBuff(int32(skill.CBSoulArrow), BuffSoulArrow)
+
+	// 2nd Job - Thief branches
+	AddSkillBuff(int32(skill.ClawBooster), BuffBooster)
+	AddSkillBuff(int32(skill.DaggerBooster), BuffBooster)
+	AddSkillBuff(int32(skill.Haste), BuffSpeed, BuffJump)
+	AddSkillBuff(int32(skill.BanditHaste), BuffSpeed, BuffJump)
+
+	// 3rd Job - Warrior branches
+	AddSkillBuff(int32(skill.ComboAttack), BuffComboAttack)
+	AddSkillBuff(int32(skill.DragonBlood), BuffWeaponAttack, BuffDragonBlood)
+	AddSkillBuff(int32(skill.DragonRoar), BuffStun)
+
+	// 3rd Job - White Knight charges
+	AddSkillBuff(int32(skill.BwFireCharge), BuffMagicAttack, BuffCharges)
+	AddSkillBuff(int32(skill.BwIceCharge), BuffMagicAttack, BuffCharges)
+	AddSkillBuff(int32(skill.BwLitCharge), BuffMagicAttack, BuffCharges)
+	AddSkillBuff(int32(skill.SwordFireCharge), BuffMagicAttack, BuffCharges)
+	AddSkillBuff(int32(skill.SwordIceCharge), BuffMagicAttack, BuffCharges)
+	AddSkillBuff(int32(skill.SwordLitCharge), BuffMagicAttack, BuffCharges)
+
+	// 3rd Job - Chief Bandit
+	AddSkillBuff(int32(skill.MesoGuard), BuffMesoGuard)
+	AddSkillBuff(int32(skill.Pickpocket), BuffPickPocketMesoUP)
+
+	// 3rd Job - Hermit
+	AddSkillBuff(int32(skill.MesoUp), BuffMesoUP)
+	AddSkillBuff(int32(skill.ShadowPartner), BuffShadowPartner)
+
+	// 3rd Job - Priest
+	AddSkillBuff(int32(skill.HolySymbol), BuffHolySymbol)
+
+	// GM skills
+	AddSkillBuff(int32(skill.GMShadowPartner), BuffShadowPartner)
+	AddSkillBuff(int32(skill.GMBless), BuffWeaponAttack, BuffWeaponDefense, BuffMagicAttack, BuffMagicDefense, BuffAccuracy, BuffAvoidability)
+	AddSkillBuff(int32(skill.GMHaste), BuffSpeed, BuffJump)
+	AddSkillBuff(int32(skill.GMHolySymbol), BuffHolySymbol)
+	AddSkillBuff(int32(skill.Hide), BuffInvincible)
+}
+
+func init() {
+	LoadBuffs()
+}
+
+// CharacterBuffs is a Go adaptation of the provided C# CharacterBuffs class.
```

Comment:
Do you have a reference to the C# codebase? We should add it to the readme acknowledgements section.